File: Php/app/Urls/BaseClasses/HomegameYearUrlModel.php
<?php
namespace app\Urls\BaseClasses{

	use app\UrlFormatter;
	use entities\Homegame;

	class HomegameYearUrlModel extends UrlModel{

		public function __construct($format, $formatWithYear, Homegame $homegame, $year){
			if($year != null){
				$this->url = UrlFormatter::formatHomegameWithYear($formatWithYear, $homegame, $year);
			} else {
				$this->url = UrlFormatter::formatHomegame($format, $homegame);
			}
		}

	}

}

File: Web/Models/Url/HomegameYearUrlModel.cs
using Core.Classes;

namespace Web.Models.Url{

	class HomegameYearUrlModel : UrlModel{

	    public HomegameYearUrlModel(string format, string formatWithYear, Homegame homegame, int? year)
	    {
            if (year.HasValue)
            {
                SetUrl(UrlFormatter.FormatHomegameWithYear(formatWithYear, homegame, year.Value));
            }
            else
            {
                SetUrl(UrlFormatter.FormatHomegame(format, homegame));
            }
	    }
	}

}